src/client/components/SearchMessagesWords/SearchMessages.js

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import './SearchMessages.css';
+
+export default function SearchMessages() {
+  const [searchTerm, setSearchTerm] = React.useState('');
+  const [searchResults, setSearchResults] = React.useState([]);
+
+  const handleInputChange = (event) => {
+    setSearchTerm(event.target.value);
+  };
+
+  useEffect(() => {
+    fetch(`/api/messages?query=${searchTerm}`)
+      .then((response) => response.json())
+      .then((data) => setSearchResults(data.searchResults));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  React.useEffect(() => {
+    const results = searchResults.filter((message) =>
+      message.toLowerCase().includes(searchTerm.toLowerCase()),
+    );
+    setSearchResults(results);
```

Comment:
By doing this you are overwriting the search results. Instead of searching already fetched messages, you need to go to the API back again.